features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.h

```diff
@@ -0,0 +1,111 @@
+/*
+  *  aes_alt.h
+  *
+  *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  *
+  */
+
+#ifndef __AES_ALT__
+#define __AES_ALT__
+
+#include "ssi_aes.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+struct sw_aes_context
+{
+    int nr;                     /*!<  number of rounds  */
+    uint32_t *rk;               /*!<  AES round keys    */
+    uint32_t buf[68];           /*!<  unaligned data    */
+    int sw_key_set;
+};
+
+typedef struct
+{
+    struct {
+        SaSiAesUserContext_t CC_Context;
+        SaSiAesEncryptMode_t CC_cipherFlag;
+        SaSiAesUserKeyData_t CC_KeyData;
+    } cc_aes_context;
+    struct sw_aes_context sw_context;
```

Comment:
correct, but if we have this overlapping merged, we will have much implementation difficulties, when we need to choose between sw\HW implementation. I think there is a need for changing some of the mbed TLS aes implementation, to support fallbacks, see https://github.com/ARMmbed/mbedtls/issues/818 for more information